finagle-http/src/main/scala/com/twitter/finagle/http/Codec.scala

```diff
@@ -521,3 +522,107 @@ private[finagle] class HttpServerTracingFilter[Req <: Request, Res](serviceName:
     service(request)
   }
 }
+
+private[finagle] object OpenTracing {
+
+  import io.opentracing.Tracer
+  import io.opentracing.Span
+  import java.util.HashMap
+  import scala.collection.JavaConverters._
+
+  /**
+   * Set the parent span of all spans created during the scope
+   * of the enclosed function.
+   */
+  def letParentSpan[R](span: Span)(f: => R): R = {
+    Contexts.local.let(spanCtx, span)(f)
+  }
+
+  /**
+   * Set the operation name for all spans created during the scope
+   * of the enclosed function.
+   */
+  def letOperationName[R](operationName: String)(f: => R): R = {
+    Contexts.local.let(rpcCtx, operationName)(f)
+  }
+
+  private[this] val spanCtx = Contexts.local.newKey[Object]
+  private[this] val rpcCtx = Contexts.local.newKey[String]
+
+  /**
+   * Creates a new span for the HTTP request and injects the necessary
+   * parameters so the server can join the trace.
+   * 
+   * The parent span and operation name of the request can be set by
+   * calling letParentSpan and letOperationName on the request
+   *
+   * @param tracer the OpenTracing tracer to create the span
+   */
+  class ClientFilter[Req <: Request, Res](tracer: Tracer)
+    extends SimpleFilter[Req, Res]
+  {
+    def apply(request: Req, service: Service[Req, Res]) = {
+      val rpcName = Contexts.local.getOrElse(rpcCtx, () => "Finagle HTTP Request")
+      val span = createSpanFromContext(tracer, rpcName)
+      var textMap = new HashMap[String,String]()
+
+      // NOTE: OpenTracing uses `inject` to propagate arbitrary states
+      // across process boundaries without tightly coupling to any particular
+      // wire format or object type.
+      tracer.inject(span, textMap)
```

Comment:
It might be preferable to implement a `com.twitter.tracing.Tracer` that's backed by `io.opentracing.Tracer` -- this way all of finagle's built-in annotations will be automatically integrated, without requiring duplication.  This also would make the opentracing backend auto-loadable as a plugin at runtime.